### ISSUE TITLE
Linux symlink error when loading Arcade (MAME) game library

### ISSUE THREAD

**Issue by BojayB** [quinton-ashley/nostlan#27]
First of all, every time i open Nostlan it requires a setup. Now that it is setup and I have roms in my arcade folder, it is neverendingly looking for arcade roms
![image](https://user-images.githubusercontent.com/41974762/95024236-b8f24400-064f-11eb-9a0c-c282c81420c8.png)
And every time i try to add a new game it indexes the library continuously. Everything looks cool but at boot I'm already having a considerable amount of problems.

**Comment by BojayB** [comment]
Your help tab is also a link to the discord, not an invite. I cant access the discord because its taking me to a channel im not in, not to an invite to get into the server.

**Comment by quinton-ashley** [comment]
Well that's not good, Nostlan shouldn't require setup every time you start it.  What operating system are you on? If you have a very large MAME arcade game library it will take a bit of time to load all the images when you start Nostlan but shouldn't be indexing the game library. For the next release, I'm going to change the "add game" feature in the game library menu to only identify new games and add them to the user's game library. I'll add a "rescan library" button for doing what Nostlan currently does when "add game" is clicked which is rescanning the whole library.

**Comment by quinton-ashley** [comment]
@BojayB Please delete your `emu/arcade/arcadeGames.json` file. Then try the prerelease version 1.15.x that I just uploaded. I think I fixed the problem! Please let me know.

https://github.com/quinton-ashley/nostlan/releases

**Comment by BojayB** [comment]
I'm about to get the latest release, I'll keep you updated if anything goes wrong or if it works great! I'm using Lubuntu 20.04 LTS (Ubuntu 20.04.1LTS). I do not have an arcadeGames.json file, so thats already taken care of.

**Comment by BojayB** [comment]
The problem was that I didn't have a Games folder in the Arcade folder, which it told me once I opened it this time. I put my roms into the games folder I created and it loaded them all in. But once it was done loading in the 23 roms I have it just shows 7/23.

![image](https://user-images.githubusercontent.com/41974762/95246743-ed4a3980-07e2-11eb-9b52-f77bb614569e.png)

I tried doing a few other things like re-scanning and adding new games, then it resulted in flashing very quickly between 7/23 and 23/23.
Also the setup every time I open is gone, which is really nice!

**Comment by BojayB** [comment]
Also, the back button in the main menu does not function.

**Comment by BojayB** [comment]
I dont have mame installed either, let me do that and see if it fixes the problem. I forgot this is a launcher and not an emulator.

**Comment by BojayB** [comment]
NOW ITS DOING THIS AGAIN LKMAO
![image](https://user-images.githubusercontent.com/41974762/95247794-52525f00-07e4-11eb-8f52-7eaeded7e771.png)
please help I dont know what I'm doing

**Comment by quinton-ashley** [comment]
When the game library loads does Nostlan only show 7 games or 23? The picture you sent is of the loading screen when it's loading the game art images not when it's scanning the game library. MAME is kind of a special emulator because by default you should have your games in the roms folder `emu/arcade/mame/roms`. Install MAME and then put your games in there, then Nostlan will make a symlinked games folder `emu/arcade/games` -> `emu/arcade/mame/roms`. Nostlan might not be able to find game art images for all your games, could you send me the list of the arcade games you have? Please send me your `emu/arcade/arcadeGames.json` and `emu/arcade/arcadeGameLibScanLog.txt`

**Comment by BojayB** [comment]
I had 23 different files for the same game, I removed them all and put the single working rom in the folder so I only have 1 instead of 23 to make this easier.
I placed the rom inside ```emu/arcade/games``` and ```emu/arcade/mame/roms```, but it keeps loading images or something? the image below, it continuously does that even after waiting for a while. I removed the ```emu/arcade/mame/roms``` folder because you said it would make that itself, so I put the rom in my default mame roms folder and its still doing that. Ill remove it from my games folder too and see what it does. It did nothing, maybe because I didnt delete the json file yet, which i did before commenting here.
![image](https://user-images.githubusercontent.com/41974762/95333916-6a28f200-087b-11eb-9ead-2d1142daee14.png)

Its a rom for JoJo's Bizarre Adventure: Heritage for the Future.
![image](https://user-images.githubusercontent.com/41974762/95329786-84f86800-0875-11eb-9ed7-00f723d7b83e.png)

This is one of the files, I cannot send the json file.
[arcadeGameLibScanLog.txt](https://github.com/quinton-ashley/nostlan/files/5340543/arcadeGameLibScanLog.txt)

**Comment by BojayB** [comment]
Ive removed the arcadegames.json and Ive chosen the game directory on the "install mame" page (which I have installed and it knows that too because when I click install mame it says I already have it, good sign) that my mame roms are in and it just does the "scraping for the box of arcade template" again.

**Comment by quinton-ashley** [comment]
Do you have mame inside the mame folder? Delete the 'games' folder and just have the rom be in roms. Nostlan should symlink the 'games' folder to 'roms' and 'images' to 'artwork'. When Nostlan starts press command+shift+i to open the console sidebar then go to the console tab and please send me a screenshot when the program gets stuck.

**Comment by BojayB** [comment]
I believe I have mame located at ```/usr/share/games/mame``` because that's where the roms folder and other folders are. I deleted the games folder inside ```/emu/arcade```, this is the same thing that happens when I boot normally, here is the console.

![image](https://user-images.githubusercontent.com/41974762/95387669-0aecd100-08bf-11eb-8297-fcf6765abc2b.png)

**Comment by quinton-ashley** [comment]
Does it fail to download the image? Does the box image exist at `/home/bojay/Documents/emu/arcade/images/_TEMPLATE_arcade/box.png` ?

**Comment by BojayB** [comment]
![image](https://user-images.githubusercontent.com/41974762/95526509-8bd0c900-09a2-11eb-9e95-796a5c104a3f.png)

**Comment by quinton-ashley** [comment]
oh wow okay, definitely a problem with that.  It should be linked to `/usr/share/games/mame/artwork`.  Try deleting all the files in the arcade folder, then start Nostlan and see if that works.  I will look into how to fix this, thank you for reporting this error!

**Comment by quinton-ashley** [comment]
Because MAME has it's roms folder in the same dir as the emulator MAME must be installed before the user can load their game library. As of the current version of Nostlan it should no longer be possible for these errors to happen for users that don't have MAME installed yet.

**Comment by quinton-ashley** [comment]
Let me know if deleting the image folder symlink/alias (not the artwork folder) and then restarting Nostlan works. Does Nostlan symlinks it correctly now?

**Comment by BojayB** [comment]
I deleted everything in the ```emu/arcade``` folder and restarted nostlan but now its complaining about me not having a games folder. I remade the games folder and put the jojoba.zip into it and it works, but it looks like this:
![image](https://user-images.githubusercontent.com/41974762/95667728-ed766c00-0b37-11eb-9e1e-45aa32f340ef.png)
so i click on the game and now it looks like this:
(it wont upload for some reason, but it looks like painted wood chippings or something, I'll upload it in the next comment if it will let me)
And when i click anywhere, it goes back to the previous screen.
Sorry it took me so long to reply, my internet stopped working for a couple of days and I just managed to fix it.

**Comment by BojayB** [comment]
![image](https://user-images.githubusercontent.com/41974762/95667865-e9e3e480-0b39-11eb-9256-640f0ecbcee8.png)

**Comment by BojayB** [comment]
![image](https://user-images.githubusercontent.com/41974762/95668310-54981e80-0b40-11eb-9c5f-b4648a26b21f.png)
I hit enter a few times on the wood screen and this is what I get

**Comment by quinton-ashley** [comment]
jojoba.zip should go in your roms folder and the games should only be a symlink to roms. Sorry Nostlan should've done this all automatically. I'll have to investigate what's going wrong.

**Comment by quinton-ashley** [comment]
Your nav menu is hidden for some reason but if you mouse over to the corners of the screen it will appear. You can access the systems menu and try other systems out. I haven't done a lot of testing on the linux version of Nostlan so I really appreciated your help!

**Comment by BojayB** [comment]
I will turn that back on, I turned on automatic hidden menu. and my rom is in the mame roms folder, but if i dont make the games folder it asks for me to locate the games folder, should i make that the mame roms folder? if so ill do that right now and see what happens. I've also noticed theres a 1.16.x update with a 1.15.x mac file and no other files for me to use, if that was on purpose obviously thats fine that was your decision, I'm just wondering if theres been an update for the appimage I'm not aware about because I cant use nostlan without updating.
POST MESSAGE MESSAGE: Ive noticed that it is a BETA release, that could be why i dont see an appimage, but again, nostlan automatically opens the wiki page and closes the application so I can no longer use it
